data-and-programming.Rmd

```diff
@@ -182,93 +204,175 @@ We see that when a function like `log()` is called on a vector `x`, a vector is
 | `x | y`  | `x` or `y`                       | `(3 > 42) | TRUE`      | `r (3 > 42) | TRUE`      |
 | `x & y`  | `x` and `y`                      | `(3 < 4) & ( 42 > 13)` | `r (3 < 4) & ( 42 > 13)` |
 
-In `R`, logical operators are vectorized. 
+In `R`, logical operators are vectorized. The following operation returns a logical that is `TRUE` for elements that are greater than 3.
 
 ```{r}
 x = c(1, 3, 5, 7, 8, 9)
+x > 3
 ```
 
+The following returns a logical that is `TRUE` for elements that are less than 3.
+
 ```{r}
-x > 3
+x = c(1, 3, 5, 7, 8, 9)
 x < 3
+```
+
+The following returns a logical that is `TRUE` for elements that are equal to 3.
+
+```{r}
+x = c(1, 3, 5, 7, 8, 9)
 x == 3
-x != 3
 ```
 
+The following returns a logical that is `TRUE` for elements that are not equal to 3.
+
 ```{r}
+x = c(1, 3, 5, 7, 8, 9)
+x != 3
+```
+
+We can also combine logical vectors with boolean operators.
+
+```{r results = 'hold'}
 x == 3 & x != 3
 x == 3 | x != 3
 ```
 
 This is extremely useful for subsetting.
 
-```{r}
+```{r results = 'hold'}
+x = c(1, 3, 5, 7, 8, 9)
 x[x > 3]
 x[x != 3]
 ```
 
-- TODO: coercion
+If you call a function with the wrong type, R will attempt to convert it to the correct type and continue. This process is known as **type coercion**. For example, suppose we wanted to know the number of entries greater than three in the vector `x`.
```

Comment:
Should be emphasizing _implicit_ vs. _explicit_ coercion in this paragraph.

e.g `2 + TRUE` vs. `as.*()`

Reply:
Probably needs to be expanded in general. How's your `R` textbook coming along?